### PR TITLE
doc: update doc for NEMU's usage in Xiangshan workflow

### PR DESCRIPTION
**Pull request by shinezyy**

### CODE REVIEW

**Comment by shinezyy** [comment]
WIP, plz don't merge

File: README.md [hunk 1]
```diff
@@ -1,29 +1,34 @@
...
+* CPU core with support of most common ISAs
   * x86
     * real mode is not supported
     * x87 floating point instructions are not supported
   * mips32
     * CP1 floating point instructions are not supported
   * riscv32
     * only RV32IM
+  * riscv64
+    * rv64gcbh currently
```
Comment (poemonsense) [inline README.md:26]: Also support K extension

File: README.md [hunk 2]
```diff
@@ -32,59 +37,224 @@ The main features of NEMU include
...
+## FAQ
+
+### Why cannot produce a checkpoint in M-mode?
+
+Read the source code of [GCPT restorer](https://github.com/OpenXiangShan/NEMU/blob/master/resource/gcpt_restore/src/restore.S)
+
+Because we restore checkpoint in M mode, and the PC of returning to user mode is stored in EPC register.
+This recovery method will break the architecture state (EPC) if the checkpoint is produced in M mode.
+In contrast, if the checkpoint is produced in S mode or U mode,
+the return process is just like a normal interrupt return, which will not break the architecture state.
```
Comment (poemonsense) [inline README.md:219]: like a normal trap return

trap is more suitable here for both exceptions and interrupts

File: README.md [hunk 3]
```diff
@@ -32,59 +37,224 @@ The main features of NEMU include
...
+```
+
+### Use NEMU as reference design
+
+#### Build reference.so
+
+To build NEMU as reference design, run
+``` bash
+make xxx-ref_defconfig
+make menuconfig  # save
```
Comment (poemonsense) [inline README.md:157]: If I remember it correctly, `make menuconfig` is only required at the first time when NEMU is downloaded (before make any_defconfig).

After that, only make xxx_defconfig is needed to safely build NEMU.

**Review by poemonsense — APPROVED** [review]
Also considering replacing all "Xiangshan" with "XiangShan"?

**Pull request merged**